source/MaterialXView/Viewer.cpp

```diff
@@ -390,15 +390,15 @@ void Viewer::loadEnvironmentLight()
     }
 
     // Look for an irradiance map using an expected filename convention.
-    mx::ImagePtr envIrradianceMap;
+    mx::ImagePtr envIrradianceMap = nullptr;
     if (!_normalizeEnvironment && !_splitDirectLight)
     {
         mx::FilePath envIrradiancePath = _envRadianceFilename.getParentPath() / IRRADIANCE_MAP_FOLDER / _envRadianceFilename.getBaseName();
         envIrradianceMap = _imageHandler->acquireImage(envIrradiancePath);
     }
 
     // If not found, then generate an irradiance map via spherical harmonics.
-    if (!envIrradianceMap)
+    if (!envIrradianceMap || envIrradianceMap == _imageHandler->getInvalidImage())
```

Comment:
This code was not being triggered since a fallback "invalid image" was being returned. Add the new api + check here to allow it to work.